lib/rules/one-var.js

```diff
@@ -249,109 +246,151 @@ module.exports = {
             return true;
         }
 
+        /**
+         * Checks a given VariableDeclaration node for errors.
+         * @param {ASTNode} node The VariableDeclaration node to check
+         * @returns {void}
+         * @private
+         */
+        function checkVariableDeclaration(node) {
+            const parent = node.parent;
+            const type = node.kind;
 
-        //--------------------------------------------------------------------------
-        // Public API
-        //--------------------------------------------------------------------------
+            if (!options[type]) {
+                return;
+            }
 
-        return {
-            Program: startFunction,
-            FunctionDeclaration: startFunction,
-            FunctionExpression: startFunction,
-            ArrowFunctionExpression: startFunction,
-            BlockStatement: startBlock,
-            ForStatement: startBlock,
-            ForInStatement: startBlock,
-            ForOfStatement: startBlock,
-            SwitchStatement: startBlock,
+            const declarations = node.declarations;
+            const declarationCounts = countDeclarations(declarations);
 
-            VariableDeclaration(node) {
-                const parent = node.parent;
-                const type = node.kind;
+            // consecutive
+            const nodeIndex = (parent.body && parent.body.length > 0 && parent.body.indexOf(node)) || 0;
```

Comment:
Will this work correctly if `parent` is a `SwitchCase`? `SwitchCase` nodes don't have a `body` property, but I assume their variable declarations should still be checked.